docs/src/almalinux_instructions.rst

```diff
@@ -0,0 +1,244 @@
+.. _almalinux_instructions:
+
+Step-by-step instructions for Almalinux 8.4
+===========================================
+
+.. note::
+
+    These instructions should also work for RHEL8, CentOS8, and Rocky Linux. Some changes may be necessary for RHEL7/CenOS7.
+
+
+1. Install Node.js
+~~~~~~~~~~~~~~~~~~
+
+carta-controller uses `Node.js <https://nodejs.org/>`_ and supports v12, v14, and v16. Node.js can easily be installed from the AlmaLinux AppStream repository. Here we install v14, as well as the ``npm`` package manager.
+
+.. code-block:: shell
+
+    # Install Node.js v14:
+    sudo dnf module enable nodejs:14
+    sudo dnf install -y nodejs npm
+
+    # Check it is installed and working:
+    node --version
+    npm --version
+
+2. Install MongoDB
+~~~~~~~~~~~~~~~~~~
+
+carta-controller uses `MongoDB <https://www.mongodb.com/>`_ to store user preferences etc. MongoDB is not available through the default AlmaLinux repositories, but we can create to custom repository file in order to easily install it.
+
+.. code-block:: shell
+    
+    # Create a custom MongoDB repo file:
+    sudo cat <<EOT >> /etc/yum.repos.d/mongodb-org.repo
+    [mongodb-org-4.4]
+    name=MongoDB Repository
+    baseurl=https://repo.mongodb.org/yum/redhat/$releasever/mongodb-org/4.4/x86_64/
+    gpgcheck=1
+    enabled=1
+    gpgkey=https://www.mongodb.org/static/pgp/server-4.4.asc
+    EOT
+
+    # Install MongoDB:
+    sudo dnf update
+    sudo dnf install -y mongodb-org
+    
+    # Start and enable MongoDB to run on startup:
+    sudo systemctl start mongod
+    sudo systemctl enable mongod
+
+    # Check that it is working
+    sudo systemctl status mongod
+
+.. note::
+
+    On RHEL7/CentOS7, MongoDB v14 can be installed as follows:
+    ``curl -fsSL https://rpm.nodesource.com/setup_14.x | bash - && yum install -y nodejs``
+
+
+3. Install carta-controller
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The easiest way to install carta-controller is using ``npm``. 
+
+.. code-block:: shell
+
+    sudo dnf install -y python3 make gcc-c++
+    sudo npm install -g --unsafe-perm carta-controller
+
+.. note::
+
+    The carta-controller executable will be installed at ``/usr/local/lib/node_modules/carta-controller``.
+    The carta-frontend will be installed at ``/usr/local/lib/node_modules/carta-controller/node_modules/carta-frontend/build``.
+
+.. note::
+    
+    Do not pass the ``--unsafe-perm`` flag to ``npm`` if using a local install.
+
+.. note::
+    Beta versions of carta-controller can be installed. For example, ``sudo npm install -g --unsafe-perm carta-controller-3.0.0-beta.1d``. 
```

Comment:
perhaps we should just say `carta-controller@beta` instead of specifying an exact version?